#64 test category api

ms214 opened the pull request:
## #️⃣연관된 이슈

#64 

## 📝작업 내용

시험 카테고리 조회/추가/삭제 기능을 추가하였습니다.
추가로 기존에 없던 카테고리에 따라 필터링 되는 기능 또한 구현하였습니다.

### 스크린샷 (선택)

## 💬리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

ms214 committed:
feat: 시험카테고리 불러오기 #64

ms214 committed:
feat: 카테고리 필터링 기능 추가 #64

#64 이슈에는 없지만 #51 이슈에서 누락된 기능 추가

ms214 committed:
feat: 카테고리 추가 api 연동, 필터링 수정 #64

필터링 기능도 수정.

ms214 committed:
feat: 카테고리 삭제 기능 추가 #64

godeka commented on src/api/queries/test/useAddCategory.js:
```diff
@@ -0,0 +1,21 @@
+
+export const useAddCategory = () =>
+  useMutation({
+    useMutation: [QUERY_KEY.addCategory],
```
mutationKey를 의도하신 건가요?

ms214 replied:
확인 후 삭제처리 하였습니다.

godeka commented on src/api/queries/test/useDeleteCategory.js:
```diff
@@ -0,0 +1,16 @@
+      console.log(payload);
+      console.log(PATH_API.DELETE_CATEGORY(payload[0]));
+      const response = await axiosInstance.delete(PATH_API.DELETE_CATEGORY(payload[0]), { data: payload[1] });
+      return response;
```
payload를 배열보다 객체로 쓰는 게 가독성이나 유지보수 측면에서 더 좋지 않을까요?

ms214 committed:
fix: payload에 배열대신 객체로 하여 코드 가독성 향상 #64

ms214 committed:
fix: comment 내용 반영

https://github.com/2024-summer-project/AcademyPro_WEB/pull/65#discussion_r1833785043

godeka reviewed (approved):
좋습니당!

ms214 committed:
Merge branch 'develop' into #64-testCategoryAPI
